pkg/operator/ceph/csi/secrets.go

```diff
@@ -42,7 +48,22 @@ const (
 	CsiCephFSProvisionerSecret          = "rook-csi-cephfs-provisioner"
 )
 
-func createCSIKeyringRBDNode(s *keyring.SecretStore) (string, error) {
+func createCSIKeyringRBDNode(context *clusterd.Context, clusterInfo *client.ClusterInfo, s *keyring.SecretStore, keepPriorCount uint32, shouldRotateCephxKeys bool) (string, error) {
+	if shouldRotateCephxKeys {
+		if keepPriorCount > 0 {
+			key, err := s.GenerateKey(appendPriorKeyCountToSecretName(csiKeyringRBDNodeUsername, keepPriorCount), cephCSIKeyringRBDNodeCaps())
+			if err != nil {
+				return "", err
+			}
+			return key, nil
+		}
+
+		err := client.AuthDelete(context, clusterInfo, csiKeyringRBDNodeUsername)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to delete older RBD node username key %s", csiKeyringRBDNodeUsername)
+		}
+	}
```

Comment:
```suggestion
		err := client.AuthDelete(context, clusterInfo, csiKeyringRBDNodeUsername)
		if err != nil {
			return "", errors.Wrapf(err, "failed to delete older RBD node username key %s", csiKeyringRBDNodeUsername)
		}
	}
```

Need to think more on this or discuss and same for similar code below

Reply:
This is the hardest part of the logic to work though I think. My previous thought has been that this high-level workflow might allow for the easiest coding. It's still a pretty rough idea though.

1. use `keyring.ShouldRotateCephxKeys()` to determine `shouldRotate` 
2. before updating keys, use `keyring.UpdatedCephxStatus()` to "peek" the future status (assuming rotation completes with success)
3. if `shouldRotate`, use the peeked keyGeneration as the "goal" (otherwise, use the current keyGeneration as the "goal")
4. pass the "goal" generation to `CreateCSISecrets()` helper functions
5. inside the helpers, if the key with the "goal" suffix doesn't exist, rotate to get to the goal (otherwise, don't rotate)
5. assuming keys rotated and rotation was successful, apply the previously peeked status